### ISSUE TITLE
Support for optional

### ISSUE THREAD

**Issue by oddball** [lifeomic/json-schema-to-graphql-types#31]
Thanks for a great tool. 

Encountered something that could be handled:

```json
    "phoneNumber": {
      "type": ["null", "string"]
    },
    "creditRatings": {
      "anyOf": [
        {
          "type": "null"
        },
        {
          "items": {
            "required": ["creditRatingAgencyId", "grade"],
``` 

That is both phoneNumber and creditRatings are optional.

**Comment by mdlavin** [comment]
Nice idea! Want to submit a fix with support for that?

**Comment by oddball** [comment]
Sorry, that was more of a humble feature request :)

**Comment by AndresXI** [comment]
Hi, I am curious if this is would be a valid JSON schema 

`{
  "id": "OptionalSchema",
  "type": "object",
  "properties": {
    "phoneNumber": {
      "type": ["null", "string"]
    },
    "creditRatings": {
      "anyOf": [{
          "type": "null"
        },
        {
          "items": {
            "required": ["creditRatingAgencyId", "grade"]
          }
        }
      ]
    }
  }
}`

**Comment by AndresXI** [comment]
Hi, I am curious if this would be a valid JSON schema: 

```
{
  "id": "OptionalSchema",
  "type": "object",
  "properties": {
    "phoneNumber": {
      "type": ["null", "string"]
    },
    "creditRatings": {
      "anyOf": [{
          "type": "null"
        },
        {
          "items": {
            "required": ["creditRatingAgencyId", "grade"]
          }
        }
      ]
    }
  }
}
```

**Comment by mdlavin** [comment]
Looks like a good start. To keep things simple you might want to handle option values with just basic types first:

```
{
  "id": "OptionalSchema",
  "type": "object",
  "properties": {
    "phoneNumber": {
      "type": ["null", "string"]
    }
  }
}
```

I'd expect the GraphQL output to look something like:

```
type OptionalSchema {
   phoneNumber: String // Note the missing ! meaning that the phoneNumber could be null
}
```

After that is working, then nested object support could be added

**Comment by AndresXI** [comment]
> Looks like a good start. To keep things simple you might want to handle option values with just basic types first:
> 
> 
> ```
> {
>   "id": "OptionalSchema",
>   "type": "object",
>   "properties": {
>     "phoneNumber": {
>       "type": ["null", "string"]
>     }
>   }
> }
> ```
> 
> I'd expect the GraphQL output to look something like:
> 
> 
> ```
> type OptionalSchema {
>    phoneNumber: String // Note the missing ! meaning that the phoneNumber could be null
> }
> ```
> 
> After that is working, then nested object support could be added

Ok, I am guessing then that we have to add a case for a null value in the mapBasicAttributeType function? 

```
function mapBasicAttributeType (type, attributeName) {
  switch (type) {
    case 'string': return GraphQLString;
    case 'integer': return GraphQLInt;
    case 'number': return GraphQLFloat;
    case 'boolean': return GraphQLBoolean;
    default: throw new Error(`A JSON Schema attribute type ${type} on attribute ${attributeName} does not have a known GraphQL mapping`);
  }
}

```

**Comment by AndresXI** [comment]
I believe every type in GraphQL is nullable by default, so we just have to take care of handling a type that includes an array such as ``` "type": ["null", "string"] ```. Not sure if I am getting somewhere with this

**Comment by mdlavin** [comment]
It's been a while since I looked at the code. The name `mapBasicAttributeType` makes me think that just basic type handling is appropriate. If you've reproduced the failure to convert and have the code you'll probably have a better idea of where the fix should go. I'd start with a testcase about how it should behave and then work backwards to find a natural way to support it.

**Comment by overhead525** [comment]
I believe I've handled the process of an array that looks like `"type": ["null", "string"]`. Now all that's left is making sure that we can handle nested objects successfully in the same manner. I have yet to test this though.

**Comment by mdlavin** [comment]
Thanks to @overhead525, a fix for this was released in 1.4.2
